api/v1alpha7/openstackmachine_types.go

```diff
@@ -84,6 +84,9 @@ type OpenStackMachineSpec struct {
 	// The volume metadata to boot from
 	RootVolume *RootVolume `json:"rootVolume,omitempty"`
 
+	// AdditionalBlockDevices is a list of specifications for additional block devices to attach to the server instance
+	AdditionalBlockDevices []AdditionalBlockDevice `json:"additionalBlockDevices,omitempty"`
```

Comment:
Please could you add
```go
// +listType=map
// +listMapKey=name
```

That last assuming you change `nameSuffix` to `name` (see below). Don't forget to re-run `make generate`, as this affects the CRDs.

This allows tools (e.g. kustomize/PATCH) to apply strategic merge patches to this list by name instead of having to replace the whole list.